sqlite/vdbemem.c

```diff
@@ -3021,6 +3026,8 @@ int sqliteVdbeMemDecimalBasicArithmetics(
   }
 
 done:
+   if( decimalfied )
+     memcpy(b, &sav, sizeof(Mem));
```

Comment:
Why do we have `don't touch these` in comment but free b->z later? Why do we need to free b->z?